incubator/command-dump/src/main/java/io/aklivity/zilla/runtime/command/dump/internal/airline/ZillaDumpCommand.java

```diff
@@ -318,253 +475,340 @@ private boolean handleFrame(
         private void onBegin(
             BeginFW begin)
         {
-            final long bindingId = begin.routedId();
-
-            if (allowedBinding.test(bindingId))
+            if (allowedBinding.test(begin.routedId()))
             {
-                final MutableDirectBuffer buffer = writeBuffer;
-                final long streamId = begin.streamId();
-                final long timestamp = begin.timestamp();
-                final long sequence = begin.sequence();
-
-                encodePcapHeader(buffer, ETHER_HEADER_SIZE + IPV6_HEADER_SIZE + TCP_HEADER_SIZE, timestamp);
-                encodeEtherHeader(buffer);
-                encodeIpv6Header(buffer, bindingId, streamId ^ 1L, streamId, TCP_HEADER_SIZE);
-                encodeTcpHeader(buffer, streamId ^ 1L, streamId, sequence - 1L, 0L, TcpFlag.SYN);
-
-                writePcapOutput(writer, buffer, 0, TCP_HEADER_LIMIT);
+                final MutableDirectBuffer buffer = new UnsafeBuffer(ByteBuffer.allocate(begin.sizeof()));
+                begin2RO = beginRW.wrap(buffer, 0, begin.sizeof()).set(begin).build();
```

Comment:
This should just be a local variable called `newBegin`.
Same feedback for other `onXXX` handlers.

Reply:
OK